### PR TITLE
Restructure devcontainer

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

With all the capabilities added to devcontainers there is no longer a need to have the Dockerfile and build that each time.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: .devcontainer.json [hunk 1]
```diff
@@ -32,8 +39,7 @@
         "[javascript]": {
           "editor.defaultFormatter": "esbenp.prettier-vscode"
         },
-        "files.trimTrailingWhitespace": true,
-        "terminal.integrated.shell.linux": "/usr/bin/zsh",
+        "terminal.integrated.defaultProfile.linux": "zsh",
```
Comment (ludeeus) [inline .devcontainer.json:42]:
```suggestion
        "files.trimTrailingWhitespace": true,
        "terminal.integrated.shell.linux": "/usr/bin/zsh",
```
unrelated change

File: .devcontainer.json [hunk 2]
```diff
@@ -1,13 +1,20 @@
 {
   "name": "Home Assistant Frontend",
-  "build": {
-    "dockerfile": "Dockerfile",
-    "context": ".."
-  },
+  "image": "mcr.microsoft.com/devcontainers/python:0-3.10",
```
Comment (ludeeus) [inline .devcontainer.json:3]: Since this is a frontend repo it might make sense to use node as the base instead of python.
But adding python ontop of node takes around 30s on my machine while adding node ontop of python takes 3s.

Comment (steverep) [reply .devcontainer.json:3]: Python is needed to run core inside the container

Comment (ludeeus) [reply .devcontainer.json:3]: Yes

File: .devcontainer.json [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
   "appPort": "8124:8123",
```
Comment (steverep) [inline .devcontainer.json:4]: Consider revising the ports as well. Using this variable is no longer recommended.

Comment (ludeeus) [reply .devcontainer.json:4]: Unrelated, and there is another PR open that do that.

**Review by steverep — COMMENTED** [review]
I am consistently getting errors trying to rebuild my container with this config.  I won't have time to investigate until later tonight or tomorrow.

**Comment by steverep** [comment]
I get the following error on Windows 10/WSL and latest Docker 4.16.2:

```
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code
= Unknown desc = error getting credentials - err: exit status 255, out: ``
```

It seems to be caused by the node.js feature, which I don't think we need anyway since the python image already installs nvm.  Just need to add a hook to install.

**Comment by ludeeus** [comment]
If you have issues with features you should fix those, those are not new, and are here to stay.

**Comment by steverep** [comment]
I don't know that it's a problem with all features.  I spent a fair amount of time trying to figure out the cause and most sources point to various docker bugs.  My docker settings are all default.

Does this work for you on Windows?

**Comment by ludeeus** [comment]
I do not use windows.
